src/welldev/io/DAO/ProductDAOImplementation.java

```diff
@@ -0,0 +1,142 @@
+package welldev.io.DAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+
+import welldev.io.model.Product;
+import welldev.io.utility.AllQueryStrings;
+import welldev.io.utility.DBConnection;
+
+public class ProductDAOImplementation implements ProductDAO{
+	
+	private static PreparedStatement preparedStatement;
+	private static Statement statement;
+	private static ResultSet resultSet;
+
+	@Override
+	public ArrayList<Product> retrieveAll() {
+		ArrayList<Product>productList = new ArrayList<>();
+		try {
+			Connection dbConnection = DBConnection.getInstance().getConnection();
+			statement = dbConnection.createStatement();
+			
+			resultSet = statement.executeQuery(AllQueryStrings.allProduct);
+			while(resultSet.next()) {
+				Product pl = new Product();
+				pl.setId(resultSet.getInt("id"));
+				pl.setName(resultSet.getString("product_name"));
+				pl.setPrice(resultSet.getFloat("price"));
+				pl.setQuantity(resultSet.getInt("quantity"));
+				productList.add(pl);
+			}
+			
+		} catch (SQLException e) {
```

Comment:
When SQLException occurred...would you please explain?

Reply:
when there is a problem with the query and information cannot be retrieved because of that.